File: 28-file-stats/file_stat.py
# Python program to display file statistics
# Written by M.V.Harish Kumar on 25-05-2024 
import sys

stats = dict.fromkeys(["lines", "words", "bytes"])
n_alphanum = 0

with open(sys.argv[1]) as file:
    ftxt = file.read()
    stats['bytes'] = len(ftxt)
    stats['words'] = len(ftxt.split())
    stats['lines'] = len(ftxt.splitlines())
    for ch in ftxt:
        if ch.isalnum():
            n_alphanum += 1

print(f"File statistics for {sys.argv[1]}")
for k, v in stats.items():
    print(f"\tNo. of {k} in file: {v}")
print("\t% of alphanumeric in file:", n_alphanum/stats['bytes']*100)

File: 28-file-stats/file_stat.exs
# Elixir program to display file statistics
# Written by M.V.Harish Kumar on 07-02-2025

[fname | _] = System.argv
ftxt = File.read! fname
stats = %{
  "bytes" => byte_size(ftxt),
  "words" => ftxt |> String.split |> length,
  "lines" => ftxt |> String.split("\n") |> length
}
n_alnum = ftxt |> String.graphemes |> Enum.filter(&(String.match?(&1, ~r/[[:alnum:]]/))) |> length
IO.puts "File statistics for #{fname}"
for {k, v} <- stats do
  IO.puts "\tNo. of #{k} in file: #{v}"
end
IO.puts "\t% of alphanumeric in file: #{n_alnum/stats["bytes"]*100}"
